library/std/src/sys/windows/path/absolute.rs

```diff
@@ -0,0 +1,76 @@
+// Original implementation taken from crate "normpath".
+// Copyright 2020 "dylni" (https://github.com/dylni)
+
+use super::super::{c, fill_utf16_buf, os2path, to_u16s, IsZero};
+use crate::io;
+use crate::path::{Path, PathBuf};
+use crate::ptr;
+
+#[cfg(test)]
+mod tests;
+
+/// Calls `GetFullPathNameW` with a workaround for incorrect results.
+pub(crate) unsafe fn get_full_path_name(
+    lpfilename: *const u16,
+    buffer: *mut u16,
+    size: c::DWORD,
+) -> c::DWORD {
+    // While the docs for `GetFullPathNameW` have the standard note about
+    // needing a `\\?\` path for a long `lpfilename`, this does not appear to
+    // be true in practice.
+    // See:
+    // https://stackoverflow.com/questions/38036943/getfullpathnamew-and-long-windows-file-paths
+    // https://googleprojectzero.blogspot.com/2016/02/the-definitive-guide-on-win32-to-nt.html
```

Comment:
After this comment was made, the docs for [`GetFullPathNameW`](https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getfullpathnamew) were updated so they no longer include the "standard note" about needing `\\?\` for long paths. So the comment is probably redundant now.

Reply:
Nice catch. I'll remove this comment.